service/src/java/org/apache/hive/service/cli/HiveSQLException.java

```diff
@@ -150,4 +189,19 @@ public static TStatus toTStatus(Exception e) {
     return tStatus;
   }
 
+  private static String addQueryId(String message, QueryState queryState) {
+    boolean opLogEnabled = HiveConf.getBoolVar(queryState.getConf(),
+        HiveConf.ConfVars.HIVE_SERVER2_LOGGING_OPERATION_ENABLED, false);
+    String opLogLevel = HiveConf.getVar(queryState.getConf(),
+        HiveConf.ConfVars.HIVE_SERVER2_LOGGING_OPERATION_LEVEL, StringUtils.EMPTY);
+    boolean asyncCompile = HiveConf.getBoolVar(queryState.getConf(),
+        HiveConf.ConfVars.HIVE_SERVER2_ASYNC_EXEC_ASYNC_COMPILE, false);
+
+    if ((!opLogEnabled || opLogLevel.equals(OperationLog.LoggingLevel.NONE.name()) || !asyncCompile) &&
```

Comment:
1. what is with `asyncCompile`? we return the operation log regardless if it's enabled?
2. why are we checking if the message doesn't contain QUERY_ID pattern? What if some exceptions do and we still want to get the operation log?